src/syntax/ast.rs

```diff
@@ -28,7 +31,6 @@ pub struct Pattern {
 
 #[derive(Debug, PartialEq)]
 pub enum Expression {
-    Pattern(Pattern),
     String(String),
     Number(Number),
```

Comment:
Please rename to `StringExpression` and `NumberExpression` as per ASDL. Also, shouldn't we define a separate `StringExpression` struct with a `value` field instead of using Rust's native string?